python/doc/source/other_systems.rst

```diff
@@ -0,0 +1,116 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. currentmodule:: pyarrow
+.. _other_systems:
+
+Using Arrow with other systems
+==============================
+
+Timestamps
+----------
+
+Timestamps are data structures that mark a particular point in time
+and we want to be able to order them, regardless of where
+they originated.
+For human consumption timestamps are usually specified by the date
+together with the time of day, often using the local time zone.
+The problem with this scheme is that if things need to be ordered
+by time across multiple time zones, using local time can be ambiguous.
+Therefore timestamps from multiple time zones should always be collected,
+stored and communicated in UTC to avoid this ambiguity. 
+
+Timestamp types in Arrow are specified with a resolution and optional
+time zone. Several utility functions exist to convert data from 
+Pandas to Arrow, including functions that convert timestamp values 
+to milliseconds, because Pandas uses nanoseconds and other systems,
+i.e. Parquet use timestamps in milliseconds.
+
+In Arrow, timestamps have two forms, depending if the time zone is
+specified or not:
+
+*   **Time zone naive** (where ``tz=None`` in Python); there is no 
+    notion of UTC or local time zone.
+
+*   **Time zone aware** where the integer values are internally 
+    normalized to UTC.
+
+Apache Spark
+++++++++++++
+
+When using Arrow with `Apache Spark <http://spark.apache.org/>`_
+special attention must be given to conversion of timestamps.
+
+Spark, unlike Arrow, does treat timestamp data without a time zone
+as being in the local time zone of the server running the code.
+Spark version 2.2 allows for setting a configuration variable:
+``spark.sql.session.timeZone`` to ``UTC``. This configuration
+variable does only work on timestamps generated with ``SparkSQL``.
+In order to force timestamps coming from other sources, i.e.
+external files to be interpreted as UTC the following code must
+be executed before the operation::
```

Comment:
I would change slightly to indicate that this needs to be done on any cluster node touching data.  maybe:
"In order to force timestamps coming from other sources, i.e. external files, to be interpreted as UTC, execute the following on all cluster nodes before the data is loaded into Python to set the system timezone::"

Reply:
Good point!